internal/clients/ec.go

```diff
@@ -91,11 +96,42 @@ func TerraformSetupBuilder(version, providerSource, providerVersion string) terr
 		ps.Configuration = map[string]interface{}{
 			envHost: ecCreds[keyHost],
 		}
-		// set environment variables for sensitive provider configuration
-		ps.Env = []string{
-			fmt.Sprintf(fmtEnvVar, envUsername, ecCreds[keyUsername]),
-			fmt.Sprintf(fmtEnvVar, envPassword, ecCreds[keyPassword]),
+
+		envVars, err := getAuthEnvVars(ecCreds)
+		if err != nil {
+			return ps, errors.Wrap(err, errExtractCredentials)
 		}
+
+		ps.Env = envVars
+
 		return ps, nil
 	}
 }
+
+func getAuthEnvVars(ecCreds map[string]string) ([]string, error) {
+	// set environment variables for sensitive provider configuration
+	// NOTE(@tnthornton) either apiKey OR username and password are valid
+	// not both.
+	apikey := ecCreds[keyAPIKey]
+	uname := ecCreds[keyUsername]
+	pword := ecCreds[keyPassword]
+
+	if apikey != "" && (uname != "" || pword != "") {
+		return nil, errors.New(errEitherAPIKeyOrUP)
+	}
```

Comment:
Is this check redundant with the subsequent two? If purpose is to inform that api key or username / pass can be used I think this error can just be moved to the uname / pass check below after the api key check

Reply:
That's fair. We can move the intent to just being informational. The checks are currently there to guard against:
* API Key defined
* Username defined

Technically we'll end up using the API Key, but it's still not a "valid" config.